src/charm.py

```diff
@@ -147,11 +144,11 @@ def _configure_kubelet(self, event):
             taints=None,
         )
 
+    @status.on_error(ops.WaitingStatus("Waiting for kube-control relation"))
```

Comment:
We wait for relations to be READY b/c the charm may be related -- but the hooks that update the data in the relation haven't yet been emitted by the juju agent. Essentially relation data take time to show up so we tell the user we're "Waiting" and they don't need to DO anything

Blocking says "I know something is missing".  Nothing is ever going to progress because you don't have a relation to a thing "like a CNI".  Blocked signals to the user that their charm won't make any progress unless they take action

Reply:
So why do we use `WaitingStatus` here? Shouldn't the user take care of the `kube-control` relation as well? https://github.com/charmed-kubernetes/charm-kubernetes-worker/pull/176/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR129

Reply:
Oh nevermind. Just saw the `_check_kubecontrol_integration` function code. All good.

Reply:
you're right maybe we don't need this check at all for `_check_kubecontrol_integration`.  it was done once during `_create_kubeconfigs` and shouldn't need to be checked again.  We could likely just remove the check and configure kube_proxy and kubelet without abandon!